services/core/shared/core/compat/sdk_v5/coreCache.js

```diff
@@ -16,38 +16,26 @@
 const { Utils, CacheLRU } = require('lisk-service-framework');
 
 const { isEmptyArray } = Utils.Data;
-const { getAccounts } = require('./coreApi');
 
 const config = require('../../../../config');
+const { knex } = require('../../../database');
 
 const cache = CacheLRU();
 const getCacheKey = (key, value) => `account:${key}:${value}`;
 
-const normalizeAccount = account => {
-	account.address = account.address.toString('hex');
-	account.token.balance = Number(account.token.balance);
-	account.sequence.nonce = Number(account.sequence.nonce);
-	account.dpos.sentVotes = account.dpos.sentVotes.map(vote => {
-		vote.delegateAddress = vote.delegateAddress.toString('hex');
-		vote.amount = Number(vote.amount);
-		return vote;
-	});
-
-	return account;
-};
-
 const getCachedAccountBy = async (key, value) => {
+	const accountsDB = await knex('accounts');
 	const cacheKey = getCacheKey(key, value);
 	let account = await cache.get(cacheKey);
 	if (!account) {
-		const result = await getAccounts({ [key]: value });
-		if (!Array.isArray(result.data) || isEmptyArray(result.data)) {
+		const result = await accountsDB.find({ [key]: value });
+		if (!Array.isArray(result) || isEmptyArray(result)) {
 			const expireMiliseconds = config.ttl.affectedByNewBlocks;
 			await cache.set(cacheKey, null, expireMiliseconds);
 			return null;
 		}
```

Comment:
`!Array.isArray(result)` would always resolve to `false` as `knex.find()` would always return an array in the response.

```suggestion
		const [result] = await accountsDB.find({ [key]: value });
		if (result === undefined) {
			const expireMiliseconds = config.ttl.affectedByNewBlocks;
			await cache.set(cacheKey, null, expireMiliseconds);
			return null;
		}
```

Reply:
Done!